src/main/java/com/github/maxopoly/MemeMana/MemeManaManager.java

```diff
@@ -1,31 +1,59 @@
 package com.github.maxopoly.MemeMana;
 
 import com.github.maxopoly.MemeMana.model.MemeManaPouch;
+import com.github.maxopoly.MemeMana.model.MemeManaUnit;
+import com.github.maxopoly.MemeMana.model.MemeManaOwner;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.UUID;
 
 public class MemeManaManager {
 
-	private Map<UUID, MemeManaPouch> playerPouches;
-	private int manaCounter;
+	private Map<Long, MemeManaPouch> playerPouches;
+	private int nextManaId;
 
 	public MemeManaManager() {
-		playerPouches = new HashMap<UUID, MemeManaPouch>();
+		playerPouches = new HashMap<Long, MemeManaPouch>();
 		reloadFromDatabase();
 	}
 
 	private void reloadFromDatabase() {
-
+		this.playerPouches = MemeManaPlugin.getInstance().getDAO().getManaPouches();
+		nextManaId = 0;
+		playerPouches.forEach((k,v) -> {
+			for(MemeManaUnit u : v.getUnits()) {
+				if(u.getID() >= nextManaId) {nextManaId = u.getID() + 1;}
```

Comment:
Should just do this in a single SQL query that selects the maximum

Reply:
My SQL power level is not high enough

Reply:
Query would just be something like ```select max(id) from manaPouches;```